### PR TITLE
Update tests for transformers 4.36

### PR DESCRIPTION
**Pull request by jenniew**
Update tests for transformers 4.36.2

Related issue: https://github.com/analytics-zoo/nano/issues/1289

Move mistral related test to main tests as mistral is working under transformers 4.36. Removed 4.34 tests

### CODE REVIEW

**Comment by jenniew** [comment]
Examples are also verified. https://github.com/analytics-zoo/nano/issues/1120

File: .github/workflows/llm_performance_tests.yml
Comment (liu-shaojun) [inline .github/workflows/llm_performance_tests.yml:27]: Please revert these changes before PR merge

Comment (jenniew) [reply .github/workflows/llm_performance_tests.yml:27]: Sure

File: python/llm/test/benchmark/arc-perf-test.yaml
Comment (hkvision) [inline python/llm/test/benchmark/arc-perf-test.yaml:19]: is this supposed to be fixed?

File: .github/workflows/llm-whisper-evaluation.yml
Comment (hkvision) [inline .github/workflows/llm-whisper-evaluation.yml:78]: revert this change to make this PR more clean.

**Comment by hkvision** [comment]
http://10.239.44.136:8888/pr_perf_gpu/ Need to check the performance before the merge.

File: python/llm/test/run-llm-inference-tests.sh
Comment (liu-shaojun) [inline python/llm/test/run-llm-inference-tests.sh:21]: Does the testing for Transformers 4.34 is no longer necessary?

File: .github/workflows/llm_performance_tests.yml
Comment (Oscilloscope98) [reply .github/workflows/llm_performance_tests.yml:27]: There seems to be other places which changed for PR tests that should be reverted

File: .github/workflows/llm_performance_tests.yml
Comment (Oscilloscope98) [inline .github/workflows/llm_performance_tests.yml:290]: The structure of igpu perf test is not corrected. I may provide some help for editing this part :)

File: python/llm/test/run-llm-inference-tests.sh
Comment (hkvision) [reply python/llm/test/run-llm-inference-tests.sh:21]: should be, mistral is put in transformers 4.36, which replaces 4.34

File: .github/workflows/llm_performance_tests.yml
Comment (Oscilloscope98) [reply .github/workflows/llm_performance_tests.yml:290]: We also need to verify igpu performance

File: .github/workflows/llm_performance_tests.yml
Comment (liu-shaojun) [inline .github/workflows/llm_performance_tests.yml:31]: For arc/core/spr perf test, except for the code that needs to be reverted, other changes LGTM. For igpu test, please refer to yuwen’s comment.

File: python/llm/test/inference/test_transformers_api.py
Comment (liu-shaojun) [inline python/llm/test/inference/test_transformers_api.py:56]: What is the reason for this change?

File: .github/workflows/llm_unit_tests.yml
Comment (liu-shaojun) [inline .github/workflows/llm_unit_tests.yml:102]: What is the reason for this change?

Comment (Oscilloscope98) [reply .github/workflows/llm_unit_tests.yml:102]: replit-code-v1-3b cannot run with transformers 4.36, so another code generation model is used here instead :)

File: python/llm/test/benchmark/arc-perf-test.yaml
Comment (jenniew) [reply python/llm/test/benchmark/arc-perf-test.yaml:19]: This is the tokenizer issue of moss-moon-003-sft model, because moss-moon-003-sft haven't fix the tokenizer issue to compatible with transformers 4.34+, we can not fix it. See the issue: https://github.com/analytics-zoo/nano/issues/1145

File: python/llm/test/inference/test_transformers_api.py
Comment (jenniew) [reply python/llm/test/inference/test_transformers_api.py:56]: replit-code-v1-3b cannot run with transformers 4.36, so another code generation model is used here instead :)

File: python/llm/test/run-llm-inference-tests.sh
Comment (jenniew) [reply python/llm/test/run-llm-inference-tests.sh:21]: yes, the testing for Transformers 4.34 is no longer necessary, since I move mistral test into general test cases.

**Comment by jenniew** [comment]
> http://10.239.44.136:8888/pr_perf_gpu/ Need to check the performance before the merge.

@hkvision @lalalapotter According to the performance report, seems mpt-7b-chat and dolly-v1 models has some performance gap. 
For mpt-7b-chat, we upgrade the model since the old model cannot be compatible with transformers 4.36, but the new model implementation has been changed. So we may need to update our optimization for the new model.
For dolly-v1, it uses transformers built-in gptj model, there has some changes between 4.31 to 4.36, which may affect the performance. But dolly-v1-6b is not seen on huggingface website now, should we still maintain this model?

File: python/llm/test/benchmark/arc-perf-test.yaml
Comment (hkvision) [reply python/llm/test/benchmark/arc-perf-test.yaml:19]: Then shall we keep transformers 4.31 in the test as well for this model?

**Review by hkvision — APPROVED** [review]
The arc related changes LGTM.

File: .github/workflows/llm_performance_tests.yml
Comment (Oscilloscope98) [inline .github/workflows/llm_performance_tests.yml:21]: We may revert the change in this file which was for PR tests

File: python/llm/test/benchmark/arc-perf-test.yaml
Comment (jenniew) [reply python/llm/test/benchmark/arc-perf-test.yaml:19]: I don't know whether we need to keep such test for transformers 4.31 since ipex-llm would be updated to support transformers 4.36. @jason-dai Do we need to keep tests which only works on transformers 4.31?

File: .github/workflows/llm_performance_tests.yml
Comment (jenniew) [reply .github/workflows/llm_performance_tests.yml]: reverted

File: .github/workflows/llm-whisper-evaluation.yml
Comment (jenniew) [reply .github/workflows/llm-whisper-evaluation.yml]: reverted

File: .github/workflows/llm_performance_tests.yml
Comment (jenniew) [reply .github/workflows/llm_performance_tests.yml]: reverted

**Review by liu-shaojun — APPROVED** [review]
LGTM

**Review by Oscilloscope98 — APPROVED** [review]

**Pull request merged**
